### ISSUE TITLE
TypeScript errors occur in markuplint development environment

### ISSUE THREAD

**Issue by apple-yagi** [markuplint/markuplint#3533]
There are numerous TypeScript errors occurring in the markuplint development environment.

If this is intentional, feel free to close this issue. However, in an era where AI-generated code has become mainstream, I believe it is important to detect these TypeScript errors as well.

<img width="1512" height="945" alt="Image" src="https://github.com/user-attachments/assets/e4c30b94-9055-4994-8b36-b8de11305e7f" />  

You can verify the errors by running the following:
```
npx tsc --noEmit
```

**Comment by YusukeHirao** [comment]
Thanks for reporting this!

### Why `npx tsc --noEmit` doesn't work in this monorepo

The root `tsconfig.json` has no `include`/`files` directive, so `npx tsc --noEmit` tries to compile the entire repository as a single compilation unit. This doesn't work correctly with our per-package project references (`composite: true`). `tsc --build --noEmit` also fails with TS6310 errors due to the same `composite` constraint.

### Why these type errors weren't caught

- `yarn build` uses `tsconfig.build.json` per package, which **excludes `*.spec.ts` files**
- `vitest` transpiles with esbuild — **no type-checking**
- Only VS Code (using `tsconfig.json`) would show these errors

### What we've done in #3543

- Fixed the type errors in `github-reporter.spec.ts` (missing required `status` property)
- Added `--typecheck` flag to `yarn test`, so **spec file type-checking is now part of the test suite**

`yarn test` now runs `vitest --typecheck run`, which performs TypeScript type-checking alongside test execution. This catches type errors in spec files without needing `npx tsc --noEmit`.

**Comment by apple-yagi** [comment]
Thank you for the detailed explanation! I was able to understand the cause of this error!
I think using vitest --typecheck is a smart approach 😄
